### PR TITLE
Show an error message if you messed up configuring this app

### PR DESCRIPTION
**Pull request by justafish**
If you set `REACT_APP_DRUPAL_BASE_URL="http://localhost:8888"` instead of `REACT_APP_DRUPAL_BASE_URL="http://127.0.0.1:8888"` you get an annoying error and nothing works

### CODE REVIEW

File: src/actions/application.js [hunk 1]
```diff
@@ -35,6 +35,9 @@ function* loadMenu() {
     yield put(resetLoading());
     yield put(showLoading());
     const menuLinks = yield call(api, 'menu');
+    if (menuLinks.message) {
+      throw new Error('Unable to access data from Drupal. Did you set REACT_APP_DRUPAL_BASE_URL to localhost instead of 127.0.0.1?');
```
Comment (dawehner) [inline src/actions/application.js]: It is always nice to provide helpful messages like this 👍 Great idea

File: src/components/06_wrappers/Default/Default.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { func, node, arrayOf, shape, string } from 'prop-types';
+import PropTypes from 'prop-types';
```
Comment (dawehner) [inline src/components/06_wrappers/Default/Default.js]: Seems like an unnecessary changes, but it also seems most people import proptypes like that.

**Comment by dawehner** [comment]
> ci/circleci: lint — Your tests failed on CircleCI
Sadly tests failed...

File: src/components/06_wrappers/Default/Default.js (hunk 2 above)
Comment (justafish) [reply src/components/06_wrappers/Default/Default.js]: yeah I was trying to figure stuff out with defaultProps to start with and it was really annoying to work with in it's current state

File: src/actions/application.js [hunk 3]
```diff
@@ -35,6 +35,11 @@ function* loadMenu() {
     yield put(resetLoading());
     yield put(showLoading());
     const menuLinks = yield call(api, 'menu');
+    if (menuLinks.message) {
```
Comment (dawehner) [inline src/actions/application.js]: I'm wondering whether we could look into the cookie and see whether the resulting response had one. ❓

File: src/reducers/application.js [hunk 4]
```diff
@@ -44,7 +44,9 @@ export default (state = initialState, action) => {
     case MENU_LOADED: {
       return {
         ...state,
-        menuLinks: action.payload.menuLinks,
+        menuLinks: Array.isArray(action.payload.menuLinks)
```
Comment (dawehner) [inline src/reducers/application.js]: How about falling back to state.menuLinks?

**Comment by lauriii** [comment]
I'm not sure how to trigger the error message. I tried to point the API to wrong url and I got following error `TypeError: Failed to fetch`.

**Comment by dawehner** [comment]
@lauriii Did you tried using 127.0.0.1 in drupal and localhost in the app configuration? That's how ran into the issue the first time by myself.

File: src/components/02_atoms/Message/Message.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { css } from 'emotion';
-import { string, oneOf } from 'prop-types';
```
Comment (mattgrill) [inline src/components/02_atoms/Message/Message.js]: Are we going to do this all over the place now? Maybe we can make an issue to migrate to this usage of `PropTypes`

File: src/reducers/application.js [hunk 6]
```diff
@@ -38,31 +38,38 @@ export default (state = initialState, action) => {
       };
     }
     case SET_MESSAGE: {
+      // This causes a new messages object to be created, instead of
+      // maintaining a reference to the old data structure.
+      const messages = state.messages.map(message => message);
```
Comment (mattgrill) [inline src/reducers/application.js]: Can we do some `...` operator magic here?

File: src/reducers/application.js (hunk 4 above)
Comment (mattgrill) [reply src/reducers/application.js]: `action.payload.menuLinks` can we just check the length?

File: src/components/02_atoms/Message/Message.js (hunk 5 above)
Comment (dawehner) [reply src/components/02_atoms/Message/Message.js]: >  Maybe we can make an issue to migrate to this usage of PropTypes

Comment (dawehner) [reply src/components/02_atoms/Message/Message.js]: https://github.com/jsdrupal/drupal-admin-ui/issues/160

File: src/reducers/application.js (hunk 6 above)
Comment (dawehner) [reply src/reducers/application.js]: Interesting idea!

File: src/reducers/application.js (hunk 4 above)
Comment (justafish) [reply src/reducers/application.js]: Things which aren't arrays can also have a .length property

File: src/reducers/application.js (hunk 6 above)
Comment (justafish) [reply src/reducers/application.js]: :+1:
